src/form-data-wrapper.js

```diff
@@ -124,15 +124,44 @@ export class PolyfillFormDataWrapper {
 }
 
 /**
- * Wrap the native FormData implementation.
+ * Wrap the native `FormData` implementation.
  *
+ * NOTE: This differs from the standard `FormData` constructor. This constructor
+ * includes a submit button if it was used to submit the `opt_form`, where
+ * the native `FormData` constructor does not include the submit button used to
+ * submit the form.
+ * {@link https://xhr.spec.whatwg.org/#dom-formdata}
  * @implements {FormDataWrapperInterface}
  */
 class NativeFormDataWrapper {
   /** @override */
   constructor(opt_form) {
     /** @private @const {!FormData} */
     this.formData_ = new FormData(opt_form);
+
+    if (!opt_form) {
+      return;
+    }
+
+    // Include the submit button to match browser default submit behavior
+    const {activeElement} = opt_form.ownerDocument;
+    const {elements} = opt_form;
+    const {length} = elements;
+    for (let i = 0; i < length; i++) {
+      const element = elements[i];
+      if (element !== activeElement) {
```

Comment:
Does the have to be the active element? Can't you post from a text input, etc?

Reply:
Yes, but the standard does not specify any behavior for `FormData` when the user submits with a non-submit-button input focused.

Reply:
Can you point where? I think any form element with a name (including a submit) should be included, not just an active submit element?

Reply:
It looks like I misread it the spec (see https://github.com/ampproject/amphtml/pull/21518#discussion_r268025677 below)

https://xhr.spec.whatwg.org/#interface-formdata

Testing in Chrome I see input `<input type="button" name="foo">` not submitted generally, and `<input type="submit" name="foo">` and `<button name="foo">` submitted if they were clicked to submit, or if they are the first submit button element in the DOM tree when the user submitted indirectly.

https://jsfiddle.net/5xq20986/